cypress/e2e/API/api_EngageSphere.cy.js

```diff
@@ -0,0 +1,142 @@
+
+            });
+    })
+
+    it('Returns the correct structure of the response of the first page of customers(i.e., customers and pageInfo properties)', () => {
+        cy.request('GET', 'http://localhost:3001/customers?page=1')
+            .then((response) => {
+                expect(response.status, 'Status code').to.eq(200);
+                expect(response.body.pageInfo.totalCustomers, 'Total customers').to.be.eq(50)
+                expect(response.body.pageInfo.totalPages, 'Total pages').to.be.eq(5)
+                expect(response.body.pageInfo.currentPage, 'Current page').to.be.eq("1")
+
+
+                response.body.customers.forEach((customer) => {
+                    expect(customer, 'Customer id').to.have.property('id').that.is.a('number');
+                    expect(customer, 'Customer name').to.have.property('name').that.is.a('string');
+                    expect(customer, 'Number of employees').to.have.property('employees').that.is.a('number');
+
+                    // Verifique se contactInfo é null ou um objeto
+                    if (customer.contactInfo !== null) {
+                        expect(customer.contactInfo).to.be.an('object');
+                        expect(customer.contactInfo, "Name for contact").to.have.property('name').to.be.eq(`${customer.contactInfo.name}`)
+                        expect(customer.contactInfo, 'Email for contact').to.have.property('email').to.be.eq(`${customer.contactInfo.email}`)
+                    }
+
+                    // Verifique se address é null ou um objeto
+                    if (customer.address !== null) {
+                        expect(customer.address).to.be.an('object');
+                        expect(customer.address, "Street").to.have.property('street').that.is.a('string');
+                        expect(customer.address, 'City').to.have.property('city').that.is.a('string');
+                        expect(customer.address).to.have.property('zipCode').that.is.a('string');
+                        expect(customer.address).to.have.property('country').that.is.a('string');
+                        expect(customer.address).to.have.property('state').that.is.a('string');
+                    }
+
+                    expect(customer, 'Company Size').to.have.property('size').that.is.a('string');
+                });
+
+            });
+    })
+
+
+    context('Filter customers by size on a single page', () => {
+        it('Filter All customers', () => {
+            cy.request('GET', 'http://localhost:3001/customers?size=All&limit=50').then((response) => {
+                expect(response.status).to.eq(200);
+                expect(response.body.pageInfo.totalCustomers).to.be.eq(50)
+                expect(response.body.pageInfo.totalPages).to.be.eq(1)
+
+
+            });
+        });
+
+        it('Filters Small size customers', () => {
+            cy.request('GET', 'http://localhost:3001/customers?size=Small&limit=50').then((response) => {
```

Comment:
Se o teste é de filtragem por tamanho (`size`) passe só o `size`. Deixe o resto para os _defaults_ da API.

```suggestion
            cy.request('GET', 'http://localhost:3001/customers?size=Small').then((response) => {
```

Reply:
Além disso, veja uma alternativa que testa todos os filtros de tamanho, com um único bloco `it`.

```js
it('filters customers by size correctly', () => {
  const sizes = ['Small', 'Medium', 'Enterprise', 'Large Enterprise', 'Very Large Enterprise']
  const limitOfEmployessPerSize = [99, 999, 9999, 49999, 999999] // Assuming that there aren't companies with more than 999999 employees in the database

  sizes.forEach((size, index) => {
    cy.request('GET', `${CUSTOMERS_API_URL}?size=${size}`).as('getSizedCustomers')

    cy.get('@getSizedCustomers')
      .its('body.customers')
      .each(customer => {
        expect(customer.size).to.eq(size)
        expect(customer.employees).to.be.lte(limitOfEmployessPerSize[index])
      })
  })
})
```